### ISSUE TITLE
Two layers of extras not correctly resolved

### ISSUE THREAD

**Issue by luhn** [jazzband/pip-tools#1542]
`celery[sqs]` depends on `kombu[sqs]` which depends on boto3, pycurl, and urllib.  However, pip-tools doesn't seem to correctly resolve `kombu[sqs]`, instead just resolving plain `kombu`

#### Environment Versions

1. Mac OS
1. Python 3.9.5
1. pip 21.3.1
1. pip-compile, version 6.4.0

#### Steps to replicate

Compile the following file:

```
celery[sqs]
```

It compiles to:

```
#
# This file is autogenerated by pip-compile with python 3.9
# To update, run:
#
#    pip-compile test.in
#
amqp==5.0.6
    # via kombu
billiard==3.6.4.0
    # via celery
celery[sqs]==5.2.1
    # via -r test.in
click==8.0.3
    # via
    #   celery
    #   click-didyoumean
    #   click-plugins
    #   click-repl
click-didyoumean==0.3.0
    # via celery
click-plugins==1.1.1
    # via celery
click-repl==0.2.0
    # via celery
kombu==5.2.2
    # via celery
prompt-toolkit==3.0.23
    # via click-repl
pytz==2021.3
    # via celery
six==1.16.0
    # via click-repl
vine==5.0.0
    # via
    #   amqp
    #   celery
    #   kombu
wcwidth==0.2.5
    # via prompt-toolkit

# The following packages are considered to be unsafe in a requirements file:
# setuptools
```

However, if you look at [Celery's SQS extras](https://github.com/celery/celery/blob/6fb4f9f62801ce33dd4dc95f79d53893d6e35208/requirements/extras/sqs.txt), you can see it depends on `kombu[sqs]`.  However, the compiled requirements only lists plain `kombu`.  If you look at [Kombu's SQS extras](https://github.com/celery/kombu/blob/b183f484de677ad0235758ff3bfdeef9fc7aff52/requirements/extras/sqs.txt), you can see it depends on `boto3`, `pycurl`, and `urllib3`, which are not present in the compiled requirements.

Explicitly adding `kombu[sqs]` to the file gives us the expected results:

```
celery[sqs]
kombu[sqs]
```

```
#
# This file is autogenerated by pip-compile with python 3.9
# To update, run:
#
#    pip-compile test.in
#
amqp==5.0.6
    # via kombu
billiard==3.6.4.0
    # via celery
boto3==1.20.21
    # via kombu
botocore==1.23.21
    # via
    #   boto3
    #   s3transfer
celery[sqs]==5.2.1
    # via -r test.in
click==8.0.3
    # via
    #   celery
    #   click-didyoumean
    #   click-plugins
    #   click-repl
click-didyoumean==0.3.0
    # via celery
click-plugins==1.1.1
    # via celery
click-repl==0.2.0
    # via celery
jmespath==0.10.0
    # via
    #   boto3
    #   botocore
kombu[sqs]==5.2.2
    # via
    #   -r test.in
    #   celery
prompt-toolkit==3.0.23
    # via click-repl
pycurl==7.44.1
    # via kombu
python-dateutil==2.8.2
    # via botocore
pytz==2021.3
    # via celery
s3transfer==0.5.0
    # via boto3
six==1.16.0
    # via
    #   click-repl
    #   python-dateutil
urllib3==1.26.7
    # via
    #   botocore
    #   kombu
vine==5.0.0
    # via
    #   amqp
    #   celery
    #   kombu
wcwidth==0.2.5
    # via prompt-toolkit

# The following packages are considered to be unsafe in a requirements file:
# setuptools
```

**Comment by richafrank** [comment]
Hi @luhn , thanks for the issue. (I was able to repro.) Based on the description at least, this sounds like a duplicate of #603 .

**Comment by luhn** [comment]
Great!  I'll keep an eye on that one.
